scipy/interpolate/_fitpack_impl.py

```diff
@@ -335,6 +335,10 @@ def splev(x, tck, der=0, ext=0):
 def splint(a, b, tck, full_output=0):
     # see the docstring of `_fitpack_py/splint`
     t, c, k = tck
+    # the full docstring defines `b` as a float
+    if not isinstance(b, (int, float)):
+        # we support 1-D arrays for b, for now
+        b = b[0]
```

Comment:
Uhm, I'm really not sure about this.  If it's a hotfix to unblock the release then sure, whatever works. 

Otherwise, there should be a better way. ISTM the need for this fix means our tests are broken with newer numpy, let's fix the tests then? 

I tried to trigger the warnings locally but did not manage so far --- what do I need to do to actually see which tests they are coming from?

Reply:
@ev-br just apply this patch and then run tests: https://github.com/scipy/scipy/pull/19034#issuecomment-1672232207

Reply:
that'll show you the warnings, you may need to do some work to isolate the test(s) involved; FWIW, for `optimize` the preference was the opposite--fix the issue at the source rather than changing a test..

Reply:
Removing that line from pytest.ini does not show any warnings for me with numpy 1.25.2 still. Removing `.pytest_cache` folder or running `$ PYTHONWARNINGS=error python dev.py test -s interpolate -v ` does not help either. What am I doing wrong?

That said, I really don't want to hold the release for this. We can just merge this as is if you prefer, and deal with the rest later.

Reply:
Run the full test suite on latest `main` with that `pytest.ini` patch--just because the changes in this PR are in `interpolate` doesn't mean the `interpolate` tests trigger the issue.

`python dev.py test -j 32`

`43233 passed, 2427 skipped, 148 xfailed, 12 xpassed, 9 warnings in 112.74s (0:01:52)`

`python dev.py test -j 32 -s interpolate`

`923 passed, 2 skipped, 1 xfailed in 3.60s`

Try running with `stats` only:

`python dev.py test -j 32 -s stats`

Yup, all the warnings are from `stats` tests:

`7933 passed, 484 skipped, 45 xfailed, 9 warnings in 16.30s`

From there, I've typically used an "elimination" strategy where I have a text file with verbose output from testsuite and start re-running tests in blocks until I isolate the warning-related test(s) in question.

Reply:
Well, I can save you some time, all the warnings come from one test: `scipy/stats/tests/test_distributions.py::TestLevyStable::test_cdf_nolan_samples`

Reply:
Thank you Tyler, much appreciated!
This patch seems to fix it, at least locally:


```diff
$ git diff scipy/stats/_levy_stable/__init__.py 
diff --git a/scipy/stats/_levy_stable/__init__.py b/scipy/stats/_levy_stable/__init__.py
index 74ca97c338..cced519d5d 100644
--- a/scipy/stats/_levy_stable/__init__.py
+++ b/scipy/stats/_levy_stable/__init__.py
@@ -1113,7 +1113,7 @@ class levy_stable_gen(rv_continuous):
                     density_x, np.real(density), k=fft_interpolation_degree
                 )
                 data_out[data_mask] = np.array(
-                    [f.integral(self.a, x_1) for x_1 in _x]
+                    [f.integral(self.a, float(x_1.squeeze())) for x_1 in _x]
                 ).reshape(data_out[data_mask].shape)
 
         return data_out.T[0]
```

Yes, this is ugly :-). The problem seems to stem from the fact that in that test `_x` is a 2D array:

```
(Pdb) p _x
array([[-3.28995271e+00],
       [ 3.28995271e+00],
       [ 1.47149782e-15]])
```